docs/en/advanced-concepts/migrating-to-appium-2.0.md

```diff
@@ -127,3 +127,81 @@ TODO
+|`$cloud:automation`|Designating a custom driver to use (see below for more info). This would override `appium:automationName` and `$cloud:automationVersion`|`{"name": "org/custom-driver", "source": "github", "package": "custom-driver"}`|
+|`$cloud:appiumPlugins`|Designating the list of plugins (and potentially versions of plugins) to be activated (see below for more info).|`["images", "universal-xml"]`|
+
+### Basic example
+
+Appium extensions (drivers and plugins) have a set of properties that specify where they can be installed from. Cloud providers are obviously under no obligation to provide support for arbitrarily specified extensions, seeing as these may represent untrusted code running in a managed environment. In the case where arbitrary extensions are not supported, the `appium:automationName`, `$cloud:automationVersion`, and `$cloud:appiumPlugins` capabilities should be sufficient. See the following JSON object representing capabilities for a session:
+
+```json
+{
+    "$cloud:appiumVersion": "2.0.0",
+    "platformName": "iOS",
+    "appium:platformVersion": "14.4",
+    "appium:deviceName": "iPhone 11",
+    "appium:app": "Some-App.app.zip",
+    "appium:automationName": "XCUITest",
+    "$cloud:automationVersion": "3.52.0",
+    "$cloud:appiumPlugins": ["images"]
+}
+```
+
+This set of capabilities requests an Appium 2.0 server supporting the XCUITest driver at version `3.52.0`, and the `images` plugin active. This set is easy for a cloud provider to verify. The cloud provider can obviously do anything it wants in response to these capabilities, including downloading Appium and driver and plugin packages on the fly, or erroring out if the versions requested are not in a supported set, or if the plugin is not supported, etc...
+
+### Extension objects
+
+Some service providers may wish to dynamically allow access to all of the features of the Appium 2.0 CLI, including downloading arbitrary drivers and plugins. To represent these extensions, we can define special JSON "extension objects", with the following keys:
+
+* `name`: the name of the extension. This would be an NPM package name (if downloading from NPM), or a git or GitHub spec (if downloading from a git server or GitHub).
+* `version`: the version of the extension, e.g., the NPM package version or Git SHA.
+* (optional) `source`: a denotation of where the extension can be downloaded from. Recommended to support the following values: `appium`, `npm`, `git`, `github`. Here, `appium` means "Appium's own official list", and should be the default value if this key is not included.
+* (optional) `package`: when downloading extensions from git or github, the NPM package name of the extension must also be provided. This is optional for non-git sources.
+
+Since each session is handled by a single driver, the `$cloud:automation` capability could be used with an extension object value to denote this driver, for example:
+
+```json
+{
+    ...,
+    "$cloud:automation": {
```

Comment:
btw, I think most case to use this parameter is non-appium official drivers. (or this can be 
If this is an official xcuitest, for example, then the capabilities should be enough with the below:

```
{
    "$cloud:appiumVersion": "2.0.0",
    "platformName": "iOS",
    "appium:platformVersion": "14.4",
    "appium:deviceName": "iPhone 11",
    "appium:app": "Some-App.app.zip",
    "appium:automationName": "XCUITest",
    "$cloud:automationVersion": "3.52.0"
}
```

So, as an example here, I think it would be nice with non-npm appium custom drivers rather than simple npm like plugin's example.

Reply:
yep, good idea